### PR TITLE
[M5/M6] Store Live RP Settings in KeyVault

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/14625923

### What this PR does / why we need it:

Initial cut of storing RP config in a KeyVault that can be fetched or updated at runtime.

### Test plan for issue:

Unit tests, although no E2E yet

### Is there any documentation that needs to be updated for this PR?

N/A yet

### CODE REVIEW

**Comment by rolandmkunkel** [comment]
How does the hive AKS kubeconfig get in the keyvault in the first place?

File: pkg/deploy/generator/resources_rp.go [hunk 1]
```diff
@@ -1284,6 +1284,22 @@ func (g *generator) rpServiceKeyvaultAccessPolicies() []mgmtkeyvault.AccessPolic
 	}
 }
 
+func (g *generator) rpConfigKeyvaultAccessPolicies() []mgmtkeyvault.AccessPolicyEntry {
+	return []mgmtkeyvault.AccessPolicyEntry{
+		{
+			TenantID: &tenantUUIDHack,
+			ObjectID: to.StringPtr("[parameters('fpServicePrincipalId')]"),
```
Comment (bennerv) [inline pkg/deploy/generator/resources_rp.go:1291]: This should be the RP MSI, not the FP SP.  You're using the RP's MSI to setup a keyvault client above.

**Review by bennerv — CHANGES_REQUESTED** [review]

File: pkg/util/liveconfig/hive.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+type hiveShard struct {
+	Kubeconfig []byte `json:"kubeconfig,omitempty"`
+}
+
+type hiveConfig struct {
+	Shards []hiveShard `json:"shards,omitempty"`
+}
+
+func (p *prod) HiveRestConfig(ctx context.Context, index int) (*rest.Config, error) {
+	secret, err := p.kv.GetSecret(ctx, HiveKeyvaultSecret)
```
Comment (ross-bryan) [inline pkg/util/liveconfig/hive.go:28]: is there anyway we can verify we have all required config defined in the keyvault before we fail on grabbing individual components of the  "live config"?

**Review by ross-bryan — CHANGES_REQUESTED** [review]
this looks like good start. a couple of comments/questions and to reiterate both ben/rolands comments regarding how are we uploading/pushing to these keyvaults? and to set the ACLs to RP MSI.

**Pull request closed without merging**